matlab/src/matlab/+arrow/+type/Type.m

```diff
@@ -48,5 +48,35 @@
           propgrp = matlab.mixin.util.PropertyGroup(proplist);
         end
     end
-    
+
+    methods
+        function tf = isequal(obj, varargin)
+
+            narginchk(2, inf);
+            tf = false;
+            
+            proxyIDs = zeros([numel(obj) numel(varargin)], "uint64");
+
+            for ii = 1:numel(varargin)
+                type = varargin{ii};
+                if ~isa(type, "arrow.type.Type") || ~isequal(size(obj), size(type))
+                    % Return early if type is not an arrow.type.Type or if
+                    % the dimensions of obj and type do not match.
+                    return;
+                end
+
+                proxies = [type(:).Proxy];
```

Comment:
I think it would be helpful to add a comment here explaining what this syntax means.